eZ/Publish/Core/Repository/UserService.php

```diff
@@ -695,25 +685,32 @@ public function updateUser(APIUser $user, UserUpdateStruct $userUpdateStruct)
             return $user;
         }
 
-        if ($userUpdateStruct->email !== null) {
-            if (!is_string($userUpdateStruct->email) || empty($userUpdateStruct->email)) {
-                throw new InvalidArgumentValue('email', $userUpdateStruct->email, 'UserUpdateStruct');
-            }
-
-            if (!ezcMailTools::validateEmailAddress($userUpdateStruct->email)) {
-                throw new InvalidArgumentValue('email', $userUpdateStruct->email, 'UserUpdateStruct');
-            }
+        // Cannot update certain fields without the password.
+        $passwordRequired = false;
+        if ($userUpdateStruct->login !== null) {
+            $passwordRequired = true;
         }
 
-        if ($userUpdateStruct->password !== null && (!is_string($userUpdateStruct->password) || empty($userUpdateStruct->password))) {
+        // If password is required yet undefined, we can throw an invalid argument exception.
+        $passwordMissing = $passwordRequired && $userUpdateStruct->password === null;
+        $passwordInvalid = $userUpdateStruct->password !== null && !$this->isValidPassword($userUpdateStruct->password);
+        if ($passwordMissing || $passwordInvalid) {
             throw new InvalidArgumentValue('password', $userUpdateStruct->password, 'UserUpdateStruct');
         }
 
-        if ($userUpdateStruct->enabled !== null && !is_bool($userUpdateStruct->enabled)) {
+        if ($userUpdateStruct->email !== null && !$this->isValidEmail($userUpdateStruct->email)) {
+            throw new InvalidArgumentValue('email', $userUpdateStruct->email, 'UserUpdateStruct');
+        }
+
+        if ($userUpdateStruct->login !== null && !$this->isValidLogin($userUpdateStruct->login)) {
```

Comment:
if user id is the same, then it should not return false, see content service in terms of remote id updates.